mtools/util/logevent.py

```diff
@@ -901,6 +1041,14 @@ def _parse_document(self):
         self._ninserted = doc[u'ninserted'] if 'ninserted' in doc else None
         self._ndeleted = doc[u'ndeleted'] if 'ndeleted' in doc else None
         self._numYields = doc[u'numYield'] if 'numYield' in doc else None
+        self._txnNumber = doc[u'txnNumber'] if 'txnNumber' in doc else None
```

Comment:
this whole snippet can probably be optimized in a better way:

all the lines can use `doc.get('x')` instead of doc['x'] if 'x' in doc else None

then, setting the attribute can be done with `self.__setattr__`

and all the attributes can be in a list and we can loop through them